socket_linux.go

```diff
@@ -173,9 +173,17 @@ func SocketGet(local, remote net.Addr) (*Socket, error) {
 }
 
 // SocketDiagTCPInfo requests INET_DIAG_INFO for TCP protocol for specified family type and return with extension TCP info.
-func SocketDiagTCPInfo(family uint8) ([]*InetDiagTCPInfoResp, error) {
+func SocketDiagTCPInfo(family uint8, states ...uint8) ([]*InetDiagTCPInfoResp, error) {
+	var state = uint32(1<<TCP_MAX_STATES - 1) // default all state
+	if len(states) > 0 {
```

Comment:
This parsing code could go into a separate utility function in this file.
Somebody pelase plan to make the change.

Thanks for keeping the backward compatibility of the functions and behavior.